docs/guidance/head-command.md

```diff
@@ -0,0 +1,146 @@
+# Specify container commands for head Pod
+Users want to execute some commands at two timings:
+
+* (1) **Before `ray start`**: Take [this slack thread](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1675378764037199) as an example, the user wants to set up some environment variables that will be used by `ray start`.
+
+* (2) **After `ray start` (RayCluster is ready)**: Take [this slack thread](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1669647595429959) as an example, the user wants to launch a Ray serve deployment when the RayCluster is ready.
+
+## Current KubeRay operator behavior for container commands
+* The current behavior for container commands is very weird, and **may be updated in the future**.
+* See [code](https://github.com/ray-project/kuberay/blob/47148921c7d14813aea26a7974abda7cf22bbc52/ray-operator/controllers/ray/common/pod.go#L301-L326) for more details.
+
+## Timing 1: Before `ray start`
+Currently, for timing (1), we can set the container's `Command` and `Args` in RayCluster specification to reach the goal.
+
+```yaml
+# ray-operator/config/samples/ray-cluster.head-command.yaml
+    rayStartParams:
+        ...
+    #pod template
+    template:
+      spec:
+        containers:
+        - name: ray-head
+          image: rayproject/ray:2.2.0
+          resources:
+            ...
+          ports:
+            ...
+          # `command` and `args` will become a part of `spec.containers.0.args` in the head Pod.
+          command: ["echo 123"]
+```
+* Running head Pod
```

Comment:
Do we also have a way to run on the worker pod? If not, we should state this explicitly somewhere in the beginning, and possibly we should get rid of this section title
```suggestion
* Running on the head Pod
```

Reply:
Good catch! Yes, we can use the same method to specify container commands on the worker Pods.

Reply:
Sweet! Then we can update the top of the section to be more general
```
# Specify container commands for head Pod
You can execute commands on the head pod at two timings:
```

Reply:
Updated. https://github.com/ray-project/kuberay/pull/912/commits/642cbd689d5c605abe23ddabe71343858a1732a9